### PR TITLE
Migrate customize widgets editor to use new preferences package

### PR DESCRIPTION
**Pull request by talldan**
## Description
Addresses some of https://github.com/WordPress/gutenberg/issues/31965, follows https://github.com/WordPress/gutenberg/pull/39084

Migrates the customize widgets editor to use the new preferences package (introduced in https://github.com/WordPress/gutenberg/pull/38873). Previously it was using interface, but the plan is now to deprecate the preferences APIs in interface in favor of this new package.

The main goal here is that nothing is different from a user perspective.

## Testing Instructions
1. Delete your WP localstorage data
2. Using `trunk` open the customize widgets editor and dismiss the welcome guide
3. Toggle some of the preferences in the More Menu to something other than the default (e.g. set Top toolbar to active)
4. Checkout this branch and rebuild
5. Reload the widgets editor
6. The welcome guide should not show and the preferences set in step 3 should still be the same (the data migration worked)
7. Change a few more preferences
8. Reload
9. The preferences should still be the same from step 7 (preference changes are still persisted)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/customize-widgets/src/components/more-menu/style.scss [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-	}
-
-	.components-dropdown-menu__menu {
-		padding: 0;
-	}
-}
-
-.components-popover.customize-widgets-more-menu__content {
-	z-index: z-index(".components-popover.customize-widgets-more-menu__content");
-}
```
Comment (noisysocks) [inline packages/customize-widgets/src/components/more-menu/style.scss:35]: I don't understand 😅 Why is all of this CSS no longer needed?

File: packages/e2e-tests/specs/widgets/customizing-widgets.test.js [hunk 2]
```diff
@@ -25,16 +25,21 @@ describe( 'Widgets Customizer', () => {
...
 		if ( isWelcomeGuideActive ) {
-			await page.evaluate( () =>
-				wp.data
-					.dispatch( 'core/interface' )
-					.toggleFeature( 'core/customize-widgets', 'welcomeGuide' )
+			await page.evaluate(
+				() =>
+					!! wp.data
+						.dispatch( 'core/preferences' )
+						.toggleFeature(
```
Comment (noisysocks) [inline packages/e2e-tests/specs/widgets/customizing-widgets.test.js]: Should this be `toggle`?

File: packages/e2e-tests/specs/widgets/customizing-widgets.test.js [hunk 3]
```diff
@@ -25,16 +25,21 @@ describe( 'Widgets Customizer', () => {
...
 		// Disable welcome guide if it is enabled.
-		const isWelcomeGuideActive = await page.evaluate( () =>
-			wp.data
-				.select( 'core/interface' )
-				.isFeatureActive( 'core/customize-widgets', 'welcomeGuide' )
+		const isWelcomeGuideActive = await page.evaluate(
+			() =>
+				!! wp.data
+					.select( 'core/preferences' )
+					.isFeatureActive( 'core/customize-widgets', 'welcomeGuide' )
```
Comment (noisysocks) [inline packages/e2e-tests/specs/widgets/customizing-widgets.test.js]: Should this be `get`?

Comment (talldan) [reply packages/e2e-tests/specs/widgets/customizing-widgets.test.js]: yes 😮

File: packages/customize-widgets/src/components/more-menu/style.scss (hunk 1 above)
Comment (noisysocks) [reply packages/customize-widgets/src/components/more-menu/style.scss:35]: Aha, the description of https://github.com/WordPress/gutenberg/pull/39113 answered my question!

**Review by noisysocks — APPROVED** [review]
Nice PR you have here 👍 :shipit:

File: packages/customize-widgets/src/components/more-menu/style.scss (hunk 1 above)
Comment (talldan) [reply packages/customize-widgets/src/components/more-menu/style.scss:35]: It hasn't really been needed for a while. It's identical to the css implemented by the base component in the interface package:
https://github.com/WordPress/gutenberg/blob/trunk/packages/interface/src/components/more-menu-dropdown/style.scss

I realised I could've deleted it for edit-widgets too, so I've made a small PR here to do that - https://github.com/WordPress/gutenberg/pull/39113

Comment (noisysocks) [reply packages/customize-widgets/src/components/more-menu/style.scss:35]: Thanks for cleaning it up!

**Pull request merged**